Core/src/org/sleuthkit/autopsy/datamodel/AbstractAbstractFileNode.java

```diff
@@ -174,11 +268,78 @@ private void removeListeners() {
      */
     private final PropertyChangeListener weakPcl = WeakListeners.propertyChange(pcl, null);
 
-    private void updateSheet() {
-        this.setSheet(createSheet());
+    /**
+     * Updates the values of the properties in the current property sheet with
+     * the new properties being passed in! Only if that property exists in the
+     * current sheet will it be applied. That way, we allow for subclasses to
+     * add their own (or omit some!) properties and we will not accidentally
+     * disrupt their UI.
+     *
+     * Race condition if not synchronized. Only one update should be applied at
+     * a time. The timing of currSheetSet.getProperties() could result in
+     * wrong/stale data being shown!
+     *
+     * @param newProps New file property instances to be updated in the current
+     *                 sheet.
+     */
+    private synchronized void updateProperty(ToggleableNodeProperty... newProps) {
```

Comment:
I think that this method is better named as updateSheet or updatePropertySheet. It calls setSheet.

I could also go with updateProperties.